### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by tamird**
See individual commits.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ readme = "./README.md"
 repository = "https://github.com/gimli-rs/gimli"
 
 [dependencies]
-fallible-iterator = { version = "0.2.0", default-features = false, optional = true }
+fallible-iterator = { version = "0.3.0", default-features = false, optional = true }
```
Comment (philipc) [inline Cargo.toml:15]: Do you need something from this version? This is part of the public API, so I prefer to leave this until we have other breaking changes.

File: examples/dwarfdump.rs [hunk 2]
```diff
@@ -497,7 +497,7 @@ fn main() {
                 process::exit(1);
             }
         };
-        let mmap_ref = (*arena_mmap.alloc(mmap)).borrow();
+        let mmap_ref = &*arena_mmap.alloc(mmap);
```
Comment (philipc) [inline examples/dwarfdump.rs:500]: These changes are good, but it seems like this is a false positive for the rust compiler warning?

File: Cargo.toml (hunk 1 above)
Comment (tamird) [reply Cargo.toml]: At one point I needed https://github.com/sfackler/rust-fallible-iterator/pull/34, but it seems I don't anymore. Removed this bump.

File: examples/dwarfdump.rs (hunk 2 above)
Comment (tamird) [reply examples/dwarfdump.rs]: Yeah, I think you're right. I've changed this a bit to simplify.

Comment (philipc) [reply examples/dwarfdump.rs]: Ah nice, even better.

**Pull request merged**